### PR TITLE
Datasetinfo s3 hierarchy

### PR DESCRIPTION
**Pull request by kotfic**
Implements a hierarchical widget that allows users to select a file from an S3 dataset.  This information is then persisted back to the server in the dataset's 'minerva' metadata field.

### CODE REVIEW

File: server/rest/s3_dataset.py [hunk 1]
```diff
@@ -122,3 +127,37 @@ def createDataset(self, item, params):
...
+        params['folderId'] = ObjectId(params['folderId'])
+
+        # Metadata keys we allow to be updated
+        valid_keys = ['selectedItems']
+
+        try:
+            minerva_metadata = item['meta']['minerva']
+        except KeyError:
+            raise RestException(u"No metadata set, is this really a dataset?")
+
```
Comment (cjh1) [inline server/rest/s3_dataset.py]: Why use unicode string here?

File: server/rest/s3_dataset.py [hunk 2]
```diff
@@ -122,3 +127,37 @@ def createDataset(self, item, params):
...
+        # Strictly allow only updates to keys in valid_keys
+        if len(set([i for i in minerva_metadata.items() if i[0] not in valid_keys]) ^
+               set([i for i in params.items() if i[0] not in valid_keys])) > 0:
+            raise RestException("Only %s may be updated!" % valid_keys)
+
```
Comment (cjh1) [inline server/rest/s3_dataset.py]: Do you think this could be simplified a little, don't we just want to check that all keys are in valid_keys?

Comment (kotfic) [inline server/rest/s3_dataset.py]: Probably could be simplified (or better named?),  the idea was to white list values that can be updated through this method and disallow all other metadata values to changed.

keys like assetstoreId and and original_type shouldn't be allowed to change correct?

**Comment by kotfic** [comment]
Locally both pep8 and jsstyle_minerva are passing - not sure why travis is showing failure?

**Comment by cjh1** [comment]
@kotfic Is this ready to merge?

**Comment by kotfic** [comment]
@cjh1 Should be ready assuming tests pass

**Comment by cjh1** [comment]
LGTM

**Pull request merged**